core-js/src/main/javascript/cdf-legacy/Dashboards.RefreshEngine.js

```diff
@@ -161,12 +166,20 @@ Dashboards.RefreshEngine = function(){// Manages periodic refresh of components
     fireRefresh : function(){
       activeTimer = null;
       var currentTime = getCurrentTime();
+      var component = null;
 
-      while(refreshQueue.length > 0 &&
-        refreshQueue[0].nextRefresh <= currentTime){
+      while(refreshQueue.length > 0 && refreshQueue[0].nextRefresh <= currentTime){
         var info = refreshQueue.shift();//pop first
         //call update, which calls processComponent
-        refreshComponent(info.component);
+        component = info.component;
+
+        //call update, which calls processComponent, but only if not updating
+        //if the component is updating we reinsert it into the queue
+        if(!Dashboards.isComponentUpdating(component)) {
+          refreshComponent(component);
+        }else {
+          insertInQueue(component);
+        }
       }
       if(refreshQueue.length > 0){
         activeTimer = setTimeout("Dashboards.refreshEngine.fireRefresh()", refreshQueue[0].nextRefresh - currentTime );//ToDo: cleaner way to call
```

Comment:
Isn't an else missing here? See https://github.com/webdetails/cdf/pull/986/files

Reply:
I don't think... the if should always be executed.

Reply:
Oh, PR wasn't linked in the story, I added it.

Reply:
https://github.com/webdetails/cdf/pull/991/files